app/views/site/_study_visualize.html.erb

```diff
@@ -6,8 +6,14 @@
       window.SCP.readOnlyToken = '<%= get_read_access_token(@study, current_user) %>'
     }
   <% end %>
-</script>
 
+  // TODO: Feature-flag this for DE, refine before making GA
+  // TODO (SCP-4321): Refine logic to enable fetching file from bucket, e.g. perhaps
+  //   by incorporating forthcoming `hasDE` attribute on study model
+  <% if current_user && current_user.feature_flag_for('differential_expression_frontend') %>
+    window.SCP.readOnlyToken = '<%= get_read_access_token(@study, current_user) %>'
+  <% end %>
```

Comment:
Let's remove this, and instead update `has_streamable_files?` (study.rb:1031)  with `|| current_user.feature_flag_for('differential_expression_frontend')` 

Later, as you suggest we can also update that method to be study-specific

Reply:
We also need a better method of getting the user's OAuth token into the page.  I would say that we have enough corner cases where we find ourselves needing it that it might belong inside the `UserProvider`.

Reply:
Updated per Devon in 743a1bf, and opened SCP-4336 per Jon.